src/EditorFeatures/CSharpTest/RemoveUnusedMembers/RemoveUnusedMembersTests.cs

```diff
@@ -0,0 +1,1475 @@
+        public async Task FieldOfNestedTypeIsRead()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    class NestedType
+    {
+        private int [|_goo|];
+
+        public int M() => _goo;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsUnused_PartialClass()
+        {
+            await TestInRegularAndScriptAsync(
+@"partial class MyClass
+{
+    private int [|_goo|];
+}",
+@"partial class MyClass
+{
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsRead_PartialClass()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"partial class MyClass
+{
+    private int [|_goo|];
+}
+partial class MyClass
+{
+    public int M() => _goo;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsRead_PartialClass_DifferentFile()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>partial class MyClass
+{
+    private int [|_goo|];
+}
+        </Document>
+        <Document>partial class MyClass
+{
+    public int M() => _goo;
+}
+        </Document>
+    </Project>
+</Workspace>");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsOnlyWritten_PartialClass_DifferentFile()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>partial class MyClass
+{
+    private int [|_goo|];
+}
+        </Document>
+        <Document>partial class MyClass
+{
+    public void M() { _goo = 0; }
+}
+        </Document>
+    </Project>
+</Workspace>",
+@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>partial class MyClass
+{
+}
+        </Document>
+        <Document>partial class MyClass
+{
+    public void M() { _goo = 0; }
+}
+        </Document>
+    </Project>
+</Workspace>");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsRead_InParens()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    public int M() => (_goo);
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedMembers)]
+        public async Task FieldIsWritten_InParens()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+    public int M() { (_goo) = 1; }
+}",
+@"class MyClass
+{
+    public int M() { (_goo) = 1; }
+}");
```

Comment:
what if it is:   ```public int M() => (_goo) = 1;```

Reply:
Added unit test

---
In reply to: [212812641](https://github.com/dotnet/roslyn/pull/29511#discussion_r212812641) [](ancestors = 212812641)